MetaMask/metamask-extension
# Use name component in signature requests

## Explanation

In the `flask` build type only, use the `Name` component to display addresses in typed V3 and V4 signature requests.

This enables the user to specify custom names for any of the signature request addresses, but also to select from a list of proposed names from multiple sources including ENS, Etherscan, Lens Protocol, and the blockchain itself.

## Screenshots

### Before

<img width="250" alt="v3_previous" src="https://github.com/MetaMask/metamask-extension/assets/19745142/5dd95cdb-76df-45b5-b211-9689f9938c72">
<img width="250" alt="v4_previous" src="https://github.com/MetaMask/metamask-extension/assets/19745142/ad33a59b-0ed9-4e6d-9430-12939fec4484">
<img width="250" alt="contract_details_previous" src="https://github.com/MetaMask/metamask-extension/assets/19745142/8d4192c4-708f-4f11-9633-362fbb82e7f8">

### After

<img width="250" alt="v3_new" src="https://github.com/MetaMask/metamask-extension/assets/19745142/a0d7f88b-08e3-4f66-a33a-051bcff64718">
<img width="250" alt="v4_new" src="https://github.com/MetaMask/metamask-extension/assets/19745142/f94e94e9-1fb2-4ec3-b040-b01741b2d58a">
<img width="250" alt="contract_details_new" src="https://github.com/MetaMask/metamask-extension/assets/19745142/41ba25ed-4d44-44d1-b1b3-ed5e54c0b5dd">

## Manual Testing Steps

- Go to test dApp.
- Click `Sign` under `Sign Typed Data V3`.
- Verify new components are used for addresses.
- Click address and save a custom name.
- Verify custom name is visible in new component.
- Repeat for `Sign Typed Data V4`.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
